vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt

```diff
@@ -116,11 +131,27 @@ class ScreenRecorderTestRunListener(
         attachmentListeners.forEach { it.onAttachment(test, Attachment(localVideoFile, AttachmentType.VIDEO)) }
     }
 
-    private suspend fun removeTestVideo(test: Test) {
-        val remoteFilePath = device.fileManager.remoteVideoForTest(test, testBatchId)
+    /**
+     * This can be called both when test times out and device unavailable
+     */
+    private suspend fun pullLastBatchVideo(remoteFilePath: String) {
+        val localVideoFile = fileManager.createFile(FileType.VIDEO, pool, device.toDeviceInfo(), testBatchId)
+        val millis = measureTimeMillis {
+            device.safePullFile(remoteFilePath, localVideoFile.toString())
+        }
+        logger.trace { "Pulling finished in ${millis}ms $remoteFilePath " }
+    }
+
+    private suspend fun removeRemoteVideo(remoteFilePath: String) {
         val millis = measureTimeMillis {
             device.fileManager.removeRemotePath(remoteFilePath)
         }
         logger.trace { "Removed file in ${millis}ms $remoteFilePath" }
     }
 }
+
+private suspend fun AndroidDevice.verifyHealthy(): Boolean {
+    return withTimeoutOrNull(Duration.ofSeconds(10)) {
+        executeShellCommand("echo quickhealthcheck")?.let { it.contains("quickhealthcheck") } ?: false
```

Comment:
What about adding timeout as a mandatory argument. Do we have a scenario where we should wait infinite for executing shell command?

Reply:
Not for this case. For all the other cases we have a specific timeout to do that. In this case we're not prepared to wait a long time till the shell responds (20 seconds by default). We just want to quickly establish if the device is healthy or not and pull the file or don't even try to pull it.